asdf/tests/test_schema.py

```diff
@@ -775,6 +775,64 @@ def test_custom_validation_with_definitions_bad(tmpdir):
             pass
 
 
+def test_custom_validation_with_external_ref_good(tmpdir):
+    custom_schema_path = helpers.get_test_data_path('custom_schema_external_ref.yaml')
+    asdf_file = os.path.join(str(tmpdir), 'out.asdf')
+
+    # This tree conforms to the custom schema
+    tree = {
+        'foo': asdf.tags.core.Software(name="Microsoft Windows", version="95")
+    }
+
+    with asdf.AsdfFile(tree, custom_schema=custom_schema_path) as ff:
+        ff.write_to(asdf_file)
+
+    with asdf.open(asdf_file, custom_schema=custom_schema_path) as ff:
```

Comment:
`as ff` is not needed here as it you don't use that in the `with` block below.  This might actually trigger a PEP8 warning (if we're checking)

Reply:
Oops, I think it would, I'll change these too